code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingFullscreenMessageDelegate.java

```diff
@@ -61,8 +63,9 @@ public void onShow(final FullscreenMessage fullscreenMessage) {
     @Override
     public void onDismiss(final FullscreenMessage fullscreenMessage) {
         final InternalMessage message = (InternalMessage) fullscreenMessage.getParent();
-        if (message != null && message.getAutoTrack()) {
-            message.track(null, MessagingEdgeEventType.IN_APP_DISMISS);
+        if (message != null) {
+            message.recordEventHistory(null, MessagingEdgeEventType.DISMISS);
```

Comment:
In the dismiss method here we don't track the interaction. The tracking is happening at the InternalMessageClass.

Is it possible to refactor and keep tracking and recording to EventHistory in the same class?
Or if not can we add some comments mentioning tracking happens inside Message.dismiss()